src/keyword.rs

```diff
@@ -138,28 +139,22 @@ impl Model for Keyword {
 
 /// Handles the `GET /keywords` route.
 pub fn index(req: &mut Request) -> CargoResult<Response> {
-    use diesel::expression::dsl::sql;
-    use diesel::types::BigInt;
     use schema::keywords;
 
     let conn = req.db_conn()?;
     let (offset, limit) = req.pagination(10, 100)?;
     let query = req.query();
     let sort = query.get("sort").map(|s| &s[..]).unwrap_or("alpha");
 
-    let mut query = keywords::table
-        .select((keywords::all_columns, sql::<BigInt>("COUNT(*) OVER ()")))
-        .limit(limit)
-        .offset(offset)
-        .into_boxed();
+    let mut query = keywords::table.into_boxed();
 
     if sort == "crates" {
         query = query.order(keywords::crates_cnt.desc());
     } else {
         query = query.order(keywords::keyword.asc());
     }
 
-    let data = query.load::<(Keyword, i64)>(&*conn)?;
+    let data = query.paginate(limit, offset).load::<(Keyword, i64)>(&*conn)?;
```

Comment:
It's a little unfortunate that selecting the total number of results is now gone from this location, but we have to still remember to include the `i64` type in the `load` type here. Not sure what to do differently though, maybe if we made a `paginated_load` that only took the types this code knows about and inserts the `i64`? 

Not a huge deal, this PR is still a huge improvement for factoring out duplication and raw SQL everywhere, just a thought for a possible next step and/or if pagination becomes a more general diesel thing.